documentation/Language/Locale.md

```diff
@@ -0,0 +1,47 @@
+# Locale
+
+This feature of the language allows specifing localization tokens and values for languages indentified by [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) codes.
+
+```mint
+locale en {
+  ui: {
+    buttons: {
+      ok: "OK"
+    }
+  }
+}
+
+locale hu {
+  ui: {
+    buttons: {
+      ok: "Rendben"
+    }
+  }
+}
+```
+
+A locale consists of a tree structure where the keys are lowercase identifiers and the values are expressions.
+
+To localize a value you need to use the locale token:
+
+```
+:ui.buttons.ok
```

Comment:
Using an `:atom` token seems IMO too generic and kinda unnecessary, I'd rather go with regular function in the global scope, sth like `t(...)`.

Reply:
There are no global functions in Mint and they need to learn that as well, also if we would use that it could conflict with any locally defined function with the same name.

We could put it on `Locale` module / store (`Locale.t`) but it would make it harder to connect the components to it in order to trigger a re-render when the locale changes https://github.com/mint-lang/mint/pull/618/files#diff-2a98a3d17ef6693f5e57ca9b02ea16a62585b9b2e038e525c9ce7ae0cb4c3c91R146 and also it would include a lot of boilerplate.

Localization is doable without a language feature, but with a lot of boilerplate:
```
// Need to define the type and subtyles for the locale
record Locale {
  navigation : Locale.Navigation,
  queryString : Locale.Language,
  ...
}

record Locale.Navigation {
  ...
}

record Locale.Language {
  ...
}

// Need to define locales in a module
module Locales {
  const EN =
    {
      queryString: ...
      navigation: ...
    }

// Need to have a store to house the current locale
store Locale {
  state locale : Locale = Locales.EN

  fun setLocale (string : String) {
    ...
  }

  const LOCALES =
    [
      {"EN", locale.languages.en},
      {"HU", locale.languages.hu},
      {"ES", locale.languages.es}
    ]
    |> Map.fromArray
}

// Connect to that store
component X {
  connect Locale exposing { locale }

  fun render {
    locale.queryString.someKey
  }
}
```

So this feature is basically the same but without the boilerplate.